git-lfs.go

```diff
@@ -1,11 +1,37 @@
 package main
 
 import (
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+
 	"github.com/github/git-lfs/commands"
 	"github.com/github/git-lfs/lfs"
 )
 
 func main() {
+	c := make(chan os.Signal)
+	signal.Notify(c)
+
+	var once sync.Once
+
+	go func() {
+		for {
+			sig := <-c
+			once.Do(lfs.ClearTempObjects)
+			switch sig {
+			case os.Interrupt, os.Kill:
+				fmt.Fprintf(os.Stderr, "\nExiting because of %q signal.\n", sig)
+				if commands.Erroring {
+					os.Exit(2)
+				}
+				os.Exit(0)
```

Comment:
POSIX states that the exit code when the process is terminated by a signal must be > 128 (128+n), so I think "Erroring" is redundant and we should just return 128 + signal number. Windows doesn't define this but since the environment is Git Bash adhering to POSIX makes sense. Either way a forcibly terminated process should never return 0.